jax/_src/scipy/signal.py

```diff
@@ -150,21 +150,31 @@ def _fftconvolve_unbatched(in1: Array, in2: Array, mode: str) -> Array:
 # Note: we do not re-use the code from jax.numpy.convolve here, because the handling
 # of padding differs slightly between the two implementations (particularly for
 # mode='same').
-def _convolve_nd(in1: Array, in2: Array, mode: str, *, precision: PrecisionLike) -> Array:
+def _convolve_nd(in1: Array, in2: Array, mode: str, boundary: str = 'fill', *,
+                 precision: PrecisionLike) -> Array:
   if mode not in ["full", "same", "valid"]:
     raise ValueError("mode must be one of ['full', 'same', 'valid']")
   if in1.ndim != in2.ndim:
     raise ValueError("in1 and in2 must have the same number of dimensions")
   if in1.size == 0 or in2.size == 0:
     raise ValueError(f"zero-size arrays not supported in convolutions, got shapes {in1.shape} and {in2.shape}.")
+  if boundary == 'wrap' and in1.ndim != 2:
+      raise NotImplementedError("boundary='wrap' is implemented only for 2D inputs")
   in1, in2 = promote_dtypes_inexact(in1, in2)
 
   no_swap = all(s1 >= s2 for s1, s2 in zip(in1.shape, in2.shape))
   swap = all(s1 <= s2 for s1, s2 in zip(in1.shape, in2.shape))
   if not (no_swap or swap):
     raise ValueError("One input must be smaller than the other in every dimension.")
 
+  shape1 = in1.shape
   shape_o = in2.shape
+  M = jnp.zeros(in1.ndim)
+  if in1.ndim > 1:
+    pad_width = max(shape1, shape_o)
```

Comment:
Is this the correct logic? I would have expected something like
```
pad_width = tuple(max(a, b) for a, b in zip(shape1, shape_o))
```
From the test file, it looks like we don't have any test shape pairs that would be sensitive to this difference – we may need to update the tests.

Reply:
Another way to spell this:
```
pad_width = tuple(map(max, shape1, shape2))
```

Reply:
Another way to spell this:
```
pad_width = tuple(map(max, shape1, shape2))
```

Reply:
Changed it as suggested